packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import '../base/io.dart';
+import '../doctor_validator.dart';
+
+const List<String> unsupportedVersions = <String>[
+  '6',
+  '7',
+  '8',
+];
+
+/// Validator to be run with `flutter doctor` to check
+/// Windows host machines if they are running supported versions.
+class WindowsVersionValidator extends DoctorValidator {
+  const WindowsVersionValidator({required ProcessManager processManager})
+      : _processManager = processManager,
+        super('Windows Version');
+
+  final ProcessManager _processManager;
+
+  /// Provide a literal string as the Regex pattern
+  /// and a string to validate and get a boolean determining
+  /// if the string has at least one match
+  static Iterable<RegExpMatch> validateString(String pattern, String str,
+      {bool multiLine = true}) {
+    final RegExp regex = RegExp(
+      pattern,
```

Comment:
I don't really think this helper method is carrying its weight. First, I would move the regex string to a top level and document there what it is intended to match. Then you can replace it with `regex.allMaches(contents)`

Reply:
Oh, I think from reading @christopherfujino's comment, the idea was to embed the version matching pattern in this method to make it easy to test. 

From reading the test cases, I think it would work OK without this as a public method. But if you're not going to specifically test it you should drop this.

Reply:
Yeah, Chris mentioned that it would be helpful to make this regex helper a method that we can unit test independently, so that we can feed it test strings as we start to encounter newer versions of the `stdout` from `systeminfo`.

Reply:
You're basically not testing this method, though, because this method itself is trivial. I would just test the regex itself

Reply:
I was thinking that this helper method would also include lines 58-61 and return a boolean. Also the rather than passing the pattern (which is actually what you want to test), that the function would receive `resultStdout`.

Then your test can look (something) like:

```
test('validateVersionString() correctly validates the output of `systeminfo`', () {
  bool result = validateVersionString('123');
  expect(result, isFalse);
  result = validateVersionString('OS Version:   8.0.0');
  // Too old
  expect(result, isFalse);
  result = validateVersionString('OS Version:     11.1.2');
  expect(result, isTrue);
});
```

Reply:
I think this approach is fine too, its the middle ground that isn't very useful

Reply:
I do agree with @jonahwilliams that the static method was very trivial but @christopherfujino what your test looks like is essentially what the entire validator was built to check.

I shied away from creating a regex method that returned a boolean because I didn't want to assume that there would only be one match within the `stdout`. The conditionals beginning on line 60 were added to ensure that the matches only returned 1 result AND that the major version was valid.

Reply:
ok, makes sense, i'm fine with you removing this function